### DISCUSSION TITLE
Variable Templating

### DISCUSSION THREAD

**Discussion by cseufert** [vimeo/psalm#8531] [General]
Is there any way to achieve something like this that Psalm can understand, rather than resorting to a class that just wraps the array to provide a `@template` method on getters and setters?

```php
class A {

  /** @var array<class-string<T>, T> **/
  public array $objCache = [] 

}
`

**Comment by orklah** [comment]
Hey!

Take a look at this issue: https://github.com/vimeo/psalm/issues/1969

There is a class-string-map type that was partially implemented. Beware, it's an undocumented feature and as Matt said it best: `I don't think I'll do anymore work on it – anyone who wants to use it is responsible for fixing any bugs they find :)` though for simple cases, it should work okay :)

**Reply by cseufert** [reply]
Yer, looks like to get it to even partially work, it needs the array wrapped in a class with a getter method to have the correct type signature.
